rskj-core/src/main/java/co/rsk/peg/LegacyErpFederation.java

```diff
@@ -0,0 +1,94 @@
+
+    public LegacyErpFederation(
+        List<FederationMember> members,
+        Instant creationTime,
+        long creationBlockNumber,
+        NetworkParameters btcParams,
+        List<BtcECKey> erpPubKeys,
+        long activationDelay,
+        ActivationConfig.ForBlock activations) {
+
+        super(members, creationTime, creationBlockNumber, btcParams, erpPubKeys, activationDelay, activations);
+
+        // Try getting the redeem script in order to validate it can be built
+        // using the given public keys and csv value
+        getRedeemScript();
+        validateRedeemScript();
+        getStandardRedeemScript();
+    }
+
+    @Override
+    public Script getRedeemScript() {
+        if (!activations.isActive(ConsensusRule.RSKIP284) &&
+            btcParams.getId().equals(NetworkParameters.ID_TESTNET)) {
+            logger.debug("[getRedeemScript] Returning hardcoded redeem script");
+            return new Script(ERP_TESTNET_REDEEM_SCRIPT_BYTES);
+        }
+
+        if (redeemScript == null) {
+            logger.debug("[getRedeemScript] Creating the redeem script from the keys");
+            redeemScript = activations.isActive(ConsensusRule.RSKIP293) ?
+                ErpFederationRedeemScriptParser.createErpRedeemScript(
+                    ScriptBuilder.createRedeemScript(getNumberOfSignaturesRequired(), getBtcPublicKeys()),
+                    ScriptBuilder.createRedeemScript(erpPubKeys.size() / 2 + 1, erpPubKeys),
+                    activationDelay
+                ) :
+                ErpFederationRedeemScriptParser.createErpRedeemScriptDeprecated(
+                    ScriptBuilder.createRedeemScript(getNumberOfSignaturesRequired(), getBtcPublicKeys()),
+                    ScriptBuilder.createRedeemScript(erpPubKeys.size() / 2 + 1, erpPubKeys),
+                    activationDelay
+                );
+        }
+
+        return redeemScript;
+    }
+
+    @Override
+    public Script getStandardRedeemScript() {
+        if (standardRedeemScript == null) {
+            // i think is not quite right to just "fix" the standardRedeemScript in the buggy fed.
+            // the real case is that the OP_CHECKMULTISIG is not before the OP_ELSE... as the method adds.
+            standardRedeemScript = ErpFederationRedeemScriptParser.extractStandardRedeemScript(
+                getRedeemScript().getChunks()
+            );
+        }
+        return standardRedeemScript;
+    }
+
+    private void validateRedeemScript() {
+        if (activations.isActive(ConsensusRule.RSKIP293) &&
+            this.redeemScript.equals(new Script(ERP_TESTNET_REDEEM_SCRIPT_BYTES))) {
```

Comment:
```suggestion
            this.getRedeemScript().equals(new Script(ERP_TESTNET_REDEEM_SCRIPT_BYTES))) {
```
This should do it right? And we shouldn't need to call `getRedeemScript` in the constructor that looks a bit weird